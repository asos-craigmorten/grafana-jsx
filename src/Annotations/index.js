import PropTypes from "prop-types";

const Annotations = ({ children, ...props }) => (
  <annotations list={children} {...props} />
);

Annotations.PropTypes = {
  children: PropTypes.arrayOf(PropTypes.object),
};

export default Annotations;
