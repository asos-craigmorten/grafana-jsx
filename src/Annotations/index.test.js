import Annotations from "./";

describe("Annotations", () => {
  describe("when used in isolation", () => {
    describe("when passed no children", () => {
      it("should render an annotations object containing an empty list", () => {
        expect(<Annotations />).toMatchSnapshot();
      });
    });

    describe("when passed children", () => {
      it("should render an annotations object containing an annotation", () => {
        expect(
          <Annotations>
            <test-annotation test-annotation-prop={"test-annotation-value"} />
          </Annotations>
        ).toMatchSnapshot();
      });
    });
  });

  describe("when used within another object", () => {
    describe("when passed no children", () => {
      it("should add an annotations object containing an empty list to the outer object", () => {
        expect(
          <test>
            <Annotations />
          </test>
        ).toMatchSnapshot();
      });
    });

    describe("when passed children", () => {
      it("should add an annotations object containing an annotation to the outer object", () => {
        expect(
          <test>
            <Annotations>
              <test-annotation test-annotation-prop={"test-annotation-value"} />
            </Annotations>
          </test>
        ).toMatchSnapshot();
      });
    });
  });
});
