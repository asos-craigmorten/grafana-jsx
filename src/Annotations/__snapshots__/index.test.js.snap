// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Annotations when used in isolation when passed children should render an annotations object containing an annotation 1`] = `
Object {
  "list": Array [
    Object {
      "test-annotation-prop": "test-annotation-value",
    },
  ],
}
`;

exports[`Annotations when used in isolation when passed no children should render an annotations object containing an empty list 1`] = `
Object {
  "list": Array [],
}
`;

exports[`Annotations when used within another object when passed children should add an annotations object containing an annotation to the outer object 1`] = `
Object {
  "annotations": Object {
    "list": Array [
      Object {
        "test-annotation-prop": "test-annotation-value",
      },
    ],
  },
}
`;

exports[`Annotations when used within another object when passed no children should add an annotations object containing an empty list to the outer object 1`] = `
Object {
  "annotations": Object {
    "list": Array [],
  },
}
`;
